### PR TITLE
129 setting properties

### PR DESCRIPTION
**Pull request by nschoe**
So this should be a good start at implementing the 'Set' function: we should be able to actually 'Get' and 'Set' DBus properties now!

I have tested with single types as well as complex / containers type. I have tried several "nested" cases, but it is entirely possible that I've left some aside. So please test and report if there are some use-cases that won't work. We'll try to fix it in the mean time.

Also, I must have messed up my branches because in the `diff` I see that I have included the changed in `index.js` concerning interface and error names checking.

So basically, I have made a function that checks an error and interface name against the DBus naming convention.
I have included this chekc in `bus.sendError()` and `bus.exportInterface()` so that it's not possible to export an interface whose name meet DBus naming requirements (or send an error whose name doesn't).

I'm sorry, it should have been in a separate branch, in a separate issue, but I messed this up somehow. I am not motivated enough to rebase, especially since this is a minor change :-)

### CODE REVIEW

**Comment by sidorares** [comment]
Thanks for this, @nschoe, and sorry it sat for nine years.

Closing because what it was opened for is done, but not because it was wrong — it was a correct diagnosis and the fix that landed matches yours.

**The `Set` half is fixed.** #299 corrected `lib/stdifaces.js`, which had been assigning a placeholder instead of reading the incoming value:

```js
} else {
  impl[propertyName] = 1234; // TODO: read variant and set property value
}
```

It now unpacks the variant from the message body, and `Get`/`Set` on an unknown property returns `org.freedesktop.DBus.Error.UnknownProperty` rather than silently succeeding. There is an integration test against a real `dbus-daemon`:

```
✔ writes a property via org.freedesktop.DBus.Properties
✔ errors on an unknown property
```

#129 is closed as fixed.

**The name-validation half is not, and I did not want it lost with this PR.** You flagged at the time that `lib/utils.js` had ended up on this branch by accident and belonged in its own change — you were right, and it is still unimplemented on `master`. `bus.exportInterface()` and `bus.sendError()` accept any string, and `lib/bus.js` still carries the comment admitting it:

```js
// Warning: errorName must respect the same rules as interface names (must contain a dot)
```

I have opened #309 for it, quoting your `isNameValid` implementation as the starting point.

Not merging this branch directly: it is nine years old and touches `lib/signature.js`, which has since been rewritten twice (#307 replaced the marshaller, #308 added caching and made parsed trees immutable), so it no longer applies. The examples it added — `client-properties.js` and `service-properties.js` — would still be welcome as a fresh PR if you are ever inclined; property usage remains under-documented.
